contracts/contracts/attacks/GatekeeperOneAttack.sol

```diff
@@ -1,11 +1,11 @@
 // SPDX-License-Identifier: MIT
 
-pragma solidity ^0.6.0;
+pragma solidity ^0.8.0;
 
 contract GatekeeperOneAttack {
 
-  constructor(address GatekeeperOneContractAddress) public {
-    bytes8 key = bytes8(uint64(uint16(tx.origin)) + 2 ** 32);
+  constructor(address GatekeeperOneContractAddress) {
+    bytes8 key = bytes8(uint64(uint16(uint160(tx.origin))) + 2 ** 32);
```

Comment:
Why was uint160 added here?

Reply:
In 0.8, we can only cast address to `uint160` among uints, previously we could cast it directly to `uint16`.